### PR TITLE
feat: LinuxFile supports direct fd descriptors

### PR DESCRIPTION
**Pull request by romange**
Also fix a bug where we override kver.kernel inside uring proactor (affects kernels 5.x)

### CODE REVIEW

File: util/fibers/uring_file.h [hunk 1]
```diff
@@ -59,14 +63,24 @@ class LinuxFile {
...
   std::error_code ReadFixed(io::MutableBytes dest, off_t offset, unsigned buf_index);
 
+  // int - io result.
   using AsyncCb = std::function<void(int)>;
 
   // Async versions of Read
   void ReadFixedAsync(io::MutableBytes dest, off_t offset, unsigned buf_index, AsyncCb cb);
   void ReadAsync(io::MutableBytes dest, off_t offset, AsyncCb cb);
+  void WriteFixedAsync(io::Bytes src, off_t offset, unsigned buf_index, AsyncCb cb);
+  void WriteAsync(io::Bytes src, off_t offset, AsyncCb cb);
```
Comment (dranikpg) [inline util/fibers/uring_file.h:73]: With those changes, can't we then just replace our IoMgr from DF with a plain LinuxFile?

Comment (dranikpg) [reply util/fibers/uring_file.h:73]: Btw, what do you think about overloads accepting UringBuf right here?

**Review by dranikpg — APPROVED** [review]

File: util/fibers/uring_file.h (hunk 1 above)
Comment (romange) [reply util/fibers/uring_file.h:73]: > With those changes, can't we then just replace our IoMgr from DF with a plain LinuxFile?

That's the plan.

Comment (romange) [reply util/fibers/uring_file.h:73]: > Btw, what do you think about overloads accepting UringBuf right here?

Why is it better?

Comment (dranikpg) [reply util/fibers/uring_file.h:73]: > Why is it better?

Just more convenient, we use fixed reads only with uring buf

**Pull request merged**